src/DataObjectImporter.php

```diff
@@ -146,6 +166,26 @@ private function loadMediaEntities(MediaTypeInterface $media_type, ObjectInterfa
     return $results;
   }
 
+  /**
+   * Load all media entities for a given mpx object.
+   *
+   * @param \Drupal\media\MediaTypeInterface $media_type
+   *   The media type to load all entities for.
+   * @param int $mpx_object_id
+   *   The mpx object id to load the associated entities for.
+   *
+   * @return \Drupal\media\Entity\Media[]
+   *   An array of existing media entities or a new media entity.
+   */
+  private function loadMediaEntitiesById(MediaTypeInterface $media_type, int $mpx_object_id): array {
+    $media_source = $this->loadMediaSource($media_type);
+    $source_field = $media_source->getSourceFieldDefinition($media_type);
+    $media_storage = $this->entityTypeManager->getStorage('media');
+    $results = $media_storage->loadByProperties([$source_field->getName() => (string) $mpx_object_id]);
```

Comment:
I'm not sure this works.

mpx "full IDs" are URIs ending with an integer, such as this example: https://github.com/Lullabot/media_mpx/blob/8.x-1.x/tests/fixtures/media-object.json#L5

So, if we're passing in just the integer component (from the `int` typehint), casting that to a string still won't have the full URI, and $results will always be empty.

Throughout the rest of the code we try to just keep the full URI as a string or Guzzle URI object - can we do that here?